guidance/library/_json.py

```diff
@@ -94,20 +108,26 @@ def _gen_json_number(lm):
 
 
 @guidance(stateless=True)
-def _gen_json_string(lm):
-    string_chars = select(
-        [
-            char_range("a", "z"),
-            char_range("A", "Z"),
-            char_range("0", "9"),
-            *[c for c in "-_' ,.!?/[]{}():;"],
-            "\\n",
-            "\\t",
-            "\\\\",
-        ],
-        recurse=True,
-    )
-    return lm + '"' + string_chars + '"'
+def _gen_json_string(
+    lm,
+    min_length: Union[int, None] = None,
+    max_length: Union[int, None] = None,
+    regex: Union[str, None] = None,
+):
+    lm += '"'
+    if regex is not None:
+        assert min_length is None and max_length is None
+        lm += gen(regex=regex)
+    elif min_length is not None and max_length is not None:
+        assert min_length >= 0 and max_length > min_length
+        lm += exactly_n_repeats(value=select(STRING_CHARS), n_repeats=min_length)
+        lm += at_most_n_repeats(value=select(STRING_CHARS), n_repeats=(max_length - min_length))
+    else:
+        lm += select(
+            STRING_CHARS,
+            recurse=True,
+        )
```

Comment:
This functionality feels reusable! I.e. `repeat(value, min, max)` with sane behaviors when min and max are unset. This logic happens in the regex code too, so you should be able to consolidate